### PR TITLE
helm: add custom labels for loki ingress

### PR DESCRIPTION
**Pull request by geowatson**
**What this PR does / why we need it**:

Custom labels for ingresses are usually addressed from values files, so we need this feature.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [x] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

**Comment by geowatson** [comment]
there are quite a few several helm upgrades gone through today, should I wait before you merge all helm-related PRs before this one or it was the last one and i can bump version to 5.8.12 safely?

**Comment by geowatson** [comment]
@JStickler @chaudum @MichelHollands could you review this one please?

File: production/helm/loki/CHANGELOG.md
Comment (geowatson) [reply production/helm/loki/CHANGELOG.md:16]: I was waiting for the approval before changing version once again, there were a lot of changes since i published this PR.

**Comment by MichelHollands** [comment]
@geowatson Thanks for the contribution. I left one comment regarding the new version that should be used. Once https://github.com/grafana/loki/pull/9898 is merged can you rebase? Once that is done I'll merge this.

File: production/helm/loki/CHANGELOG.md
Comment (MichelHollands) [reply production/helm/loki/CHANGELOG.md:16]: The other PR has been merged. Please merge main into here and fix any issues.

**Comment by geowatson** [comment]
@MichelHollands my rebase went a little bit harsh, so i had to fully reset to main head first. Do you remember which CLI command is used to update helm values docs? The pipeline should have failed because I updated values.yaml but it haven't, so i don't remember which CMD I used for docs to re-generate.

**Review by MichelHollands — APPROVED** [review]

**Comment by MichelHollands** [comment]
@geowatson Seems like you changed enough because the CI ran without a problem. It's merged now. Thanks again.

**Comment by geowatson** [comment]
@MichelHollands checked again, https://github.com/geowatson/loki/blob/main/docs/sources/setup/install/helm/reference.md
this doc is missing ingress.labels value, which should be autogenerated here.

**Pull request merged**
